sys/include/ztimer.h

```diff
@@ -340,6 +340,20 @@ typedef struct {
      * @param   clock       ztimer clock to cancel a pending alarm, if any
      */
     void (*cancel)(ztimer_clock_t *clock);
+
+#if MODULE_ZTIMER_ONDEMAND || DOXYGEN
```

Comment:
;Let's do some bike shedding: Why not `ztimer_pm` instead of `ztimer_ondemand`?

Reply:
I think *ondemand* is a little more verbose what this pseudo module actually does. Boards making use of the driver would pull in `ztimer_ondemand_timer` if `periph_timer` shall be turned on and off. The interaction with power management will only happen indirectly through `periph_timer`.

But of course this is only my point of view. I'm completely open to other suggestions! I don't have any strong feelings regarding the name ;-)